pkg/core/transactor/listener.go

```diff
@@ -79,6 +78,17 @@ func (t *Transactor) handleSendBidTx(req *node.BidRequest) (*node.TransactionRes
 		return nil, errWalletNotLoaded
 	}
 
+	// Are we allowed to send a transaction?
+	// TODO: use a parent context
+	progress, err := t.getSyncProgress()
+	if err != nil {
+		return nil, err
+	}
+
+	if progress.Progress != 100 {
```

Comment:
In case we are one block behind due to network latency, Progress will be less than 100. In that case, no transactions would be allowed. The grpc client might need to implement `retry` mechanism in order to ensure tx was accepted by mempool.

I would usually consider implementing these restrictions at `mempool` level but not on `Transactor` level. But in case you have @autholykos vote on this Transactor-restriction design, it's OK for me.

Reply:
You're right in the fact that there could be a potential disruption if the chain is in the middle of accepting a block. However, it is incredibly difficult to move this restriction elsewhere. For instance, how will a remote node know the sync status of another node? This makes it quite infeasible to put this in the mempool for instance.

Reply:
Changes from #893 might prevent this race, if we end up going for a more unified `Chain` design. This will ensure that a sync progress percentage is never given out in the middle of block acceptance. If, however, we decide not to move forward with that, a `retry` mechanism may be needed.

Reply:
As an example, test harness already is reporting a lot of `can not send transactions when node is not synchronized`
```
    localnet_test.go:262: Node 2 sending a Bid transaction
    localnet_test.go:265: rpc error: code = Unknown desc = can not send transactions when node is not synchronized
    localnet_test.go:262: Node 3 sending a Bid transaction
    localnet_test.go:262: Node 4 sending a Bid transaction
    localnet_test.go:262: Node 5 sending a Bid transaction
    localnet_test.go:265: rpc error: code = Unknown desc = can not send transactions when node is not synchronized
    localnet_test.go:262: Node 6 sending a Bid transaction
    localnet_test.go:265: rpc error: code = Unknown desc = can not send transactions when node is not synchronized
    localnet_test.go:262: Node 7 sending a Bid transaction
    localnet_test.go:262: Node 8 sending a Bid transaction
    localnet_test.go:273: Node 0 sending a Stake transaction
    localnet_test.go:273: Node 2 sending a Stake transaction
    localnet_test.go:276: rpc error: code = Unknown desc = can not send transactions when node is not synchronized
    localnet_test.go:273: Node 5 sending a Stake transaction
    localnet_test.go:276: rpc error: code = Unknown desc = can not send transactions when node is not synchronized
    localnet_test.go:273: Node 6 sending a Stake transaction
    localnet_test.go:276: rpc error: code = Unknown desc = can not send transactions when node is not synchronized
    localnet_test.go:283: Node 2 sending a Transfer transaction
    localnet_test.go:286: rpc error: code = Unknown desc = can not send transactions when node is not synchronized
```

Reply:
This may well be because this test allows nodes to start sending transactions right as they start up - which is exactly the thing this PR is supposed to prevent. If a node is at height 0 it should be impossible to start sending transactions right away, as it was the cause of devnet crashing before.

Reply:
Summarizing the offline discussion, restricting the transactions when a node is synching up might be good practice. However, the problem observed has to do with the fact that the stake transaction has an immediate effect on committee-creation. This violates the protocol requirement, which includes a grace period, and should be corrected. Any transaction restriction optimization will be tackled in a separate issue

Reply:
Resolved with adding #901